src/utils/flatten.js

```diff
@@ -14,28 +14,70 @@ export const objToCss = (obj: Object, prevKey?: string): string => {
 }` : css
 }
 
-const flatten = (chunks: Array<Interpolation>, executionContext: ?Object): Array<Interpolation> => (
-  chunks.reduce((ruleSet: Array<Interpolation>, chunk: ?Interpolation) => {
-    /* Remove falsey values */
-    if (chunk === undefined || chunk === null || chunk === false || chunk === '') return ruleSet
-    /* Flatten ruleSet */
-    if (Array.isArray(chunk)) return [...ruleSet, ...flatten(chunk, executionContext)]
-
-    /* Handle other components */
-    // $FlowFixMe not sure how to make this pass
-    if (chunk.hasOwnProperty('styledComponentId')) return [...ruleSet, `.${chunk.styledComponentId}`]
-
-    /* Either execute or defer the function */
-    if (typeof chunk === 'function') {
-      return executionContext
-        ? ruleSet.concat(...flatten([chunk(executionContext)], executionContext))
-        : ruleSet.concat(chunk)
+const innerFlatten = (
+  targetChunks: Array<Interpolation>,
+  chunk: Interpolation,
+  executionContext: ?Object,
+) => {
+  if (chunk === undefined || chunk === null || chunk === false || chunk === '') {
+    return
+  }
+
+  const chunkType = typeof chunk
```

Comment:
I've actually never checked the perf of this. Do you know if v8 optimises typeof with some special logic? And if so will storing it in a variable break that optimisation?